ui/src/features/stage/create-stage.tsx

```diff
@@ -84,7 +86,16 @@ export const CreateStage = ({
   const [tab, setTab] = useState('wizard');
 
   const { mutateAsync, isPending } = useMutation(createResource, {
-    onSuccess: () => close()
+    onSuccess: (response) => {
+      for (const result of response?.results || []) {
+        if (result?.result?.case === 'createdResourceManifest') {
+          queryCache.stage.add(project || '', [
+            decodeUint8ArrayYamlManifestToJson(result?.result?.value)
+          ]);
+        }
+      }
```

Comment:
Same here? We have a watcher for Stages.

Reply:
Will update. Same reason as [this](https://github.com/akuity/kargo/pull/2639#discussion_r1786407864)